File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-5.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        hashSet = set()
        l = 0
        maxL = 0

        for r in range(len(s)):
            while s[r] in hashSet:
                hashSet.remove(s[l])
                l += 1
            hashSet.add(s[r])
            maxL = max(maxL, r - l + 1)
        return maxL

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-5.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxP = 0, l = 0, r = 1, n = prices.size();
        while(r < n){
            maxP = max(maxP, prices[r]-prices[l]);
            if (prices[r] < prices[l]){
                l = r;
            }
            r++;
        }
        return maxP;
    }
};
